### PR TITLE
minor improvements to creating page

### PR DESCRIPTION
**Pull request by mattcasey**
I am not able to create pages in my test space - this whole process is timing out. BUt it is because resolvePageTree takes up to 5 seconds when you have 8000 pages, and the subsequent queries are using the same transaction which times out.

I do'nt have a fix for that case, but while debugging I cleaned up some of the code

### CODE REVIEW

File: lib/permissions/pages/triggers/page-created.ts [hunk 1]
```diff
@@ -12,73 +12,65 @@ import { upsertPermission } from '../actions/upsert-permission';
...
+        createdBy: true
+      }
+    });
     if (!page) {
       throw new PageNotFoundError(pageId);
     }
 
     // This is a root page, so we can go ahead
     if (!page.parentId) {
-      // Space id could be null
```
Comment (mattcasey) [inline lib/permissions/pages/triggers/page-created.ts:23]: "Space id could be null" should never be null, according to our db. even stranger, though, the 'else' condition was still using page.spaceId in the call to upsertPermission()

File: lib/permissions/pages/triggers/page-created.ts [hunk 2]
```diff
@@ -12,73 +12,65 @@ import { upsertPermission } from '../actions/upsert-permission';
...
-            {
-              permissionLevel: 'view',
-              public: true
-            },
-            undefined,
-            tx
-          );
-        }
-      } else {
+      const permissionLevel = space?.defaultPagePermissionGroup || 'full_access';
```
Comment (mattcasey) [inline lib/permissions/pages/triggers/page-created.ts:42]: removed an if/else by just defining this variable conditionally

File: lib/permissions/pages/actions/upsert-permission.ts [hunk 3]
```diff
@@ -340,17 +339,15 @@ export async function upsertPermission(
...
-      });
+      const childrenIds = resolvedPageTree.flatChildren.map((page) => page.id);
 
       await tx.pagePermission.updateMany({
         where: {
           AND: [
             {
-              OR: childrenIds
+              pageId: {
+                in: childrenIds
```
Comment (motechFR) [inline lib/permissions/pages/actions/upsert-permission.ts:349]: good db optimisation there 👍

File: lib/permissions/pages/triggers/page-created.ts [hunk 4]
```diff
@@ -12,73 +12,65 @@ import { upsertPermission } from '../actions/upsert-permission';
 export async function setupPermissionsAfterPageCreated(pageId: string): Promise<IPageWithPermissions> {
   // connect to page node from use pages (Removing iPageWithPermissions)
   return prisma.$transaction(async (tx) => {
-    const page = await getPage(pageId, undefined, tx);
-
+    const page = await tx.page.findUnique({
+      where: {
+        id: pageId
+      },
+      select: {
```
Comment (motechFR) [inline lib/permissions/pages/triggers/page-created.ts:19]: Another good one, we get to drop the page content and permissions join query

File: lib/permissions/pages/triggers/page-created.ts (hunk 1 above)
Comment (motechFR) [reply lib/permissions/pages/triggers/page-created.ts:23]: Probably a leftover from earliest iteration where we didn't yet have the spaces enforced I think. Glad we could get rid of this

**Review by motechFR — APPROVED** [review]
Tested locally and works fine. Both a root page and a child page took between 1.3s and 1.7s in a space with 10 000 pages

**Pull request merged**